aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/usecases/ConfirmResetPasswordUseCase.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.usecases
+
+import aws.sdk.kotlin.services.cognitoidentityprovider.CognitoIdentityProviderClient
+import com.amplifyframework.auth.AuthException
+import com.amplifyframework.auth.AuthException.REPORT_BUG_TO_AWS_SUGGESTION
+import com.amplifyframework.auth.cognito.CognitoAuthExceptionConverter
+import com.amplifyframework.auth.options.AWSCognitoAuthConfirmResetPasswordOptions
+import com.amplifyframework.auth.options.AuthConfirmResetPasswordOptions
+import com.amplifyframework.core.Action
+import com.amplifyframework.core.Consumer
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+/**
+ * UseCase to run after [ResetPasswordUseCase] to respond to any challenges that service may request.
+ * [Request API](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_ConfirmForgotPassword.html)
+ */
+internal class ConfirmResetPasswordUseCase(
+    private val cognitoIdentityProviderClient: CognitoIdentityProviderClient,
```

Comment:
Since we are extracting client and config from env, should we pass env to the use cases instead?

Reply:
That would mean that use cases will need to know "how" to get them and will be impacted if env is refactored or the clients providers change. Use cases don't care how client is provided, they are just dependent on them.
Now I think about it, we could have provided either or both the `appClientId` and identity provider client as the method argument instead of constructor params. But we can take it as future improvements.

Reply:
All our actions are dependent on env anyways, since UseCases are similar to actions I feel we can make them consistent by adding env to constructor. Thoughts?